### ISSUE TITLE
Chromium rendering page very slow when Macro compositing is enabled after upgrading to Mate 1.20

### ISSUE THREAD

**Issue by darkgeek** [mate-desktop/marco#390]
#### Expected behaviour
Chromium should render page as normal.

#### Actual behaviour
I have a Samsung Chromebook Plus, which is an aarch64 device and has only llvmpipe render support. I installed Archlinux Arm, and **mate-desktop** package. When Macro compositing is enabled, Chromium renders its pages very slow, I can see "vertical refresh" clearly. However, if I disable compositing, Chromium renders fast as normal. I can confirm no such issue exists when I use Xfce4 or Mate before 1.20.

#### Steps to reproduce the behaviour
1. Get a device using llvmpipe to render
2. Enable compositing in Mate desktop
3. Open Chromium/Chrome
4. Open a web page

#### MATE general version
1.20

#### Package version
mate-desktop 1.20.0-1

#### Linux Distribution
ArchLinuxArm

Feel free to tell me if you need more information. Thanks.

**Comment by lukefromdc** [comment]
While I don't have Chromium installed, I can say that marco's compositing does not used OpenGL and thus should not interact in any way with LLVMpipe. Sounds like Chromium must be trying to use gpu acceleration (Firefox can also use it). 

This sounds like Chromium is by default using whether or not the window manager is a compositor to figure out of hardware (gpu) acceleration is available. If so, this is a corner case but not actually a bug.  Sounds like the Chrome devs are ignoring the case of CPU compositing with no GPU acceleration available, which very few desktop environments offer.  Is there a preference in Chromium to turn gpu acceleration off? If so, try it.

Thus, this is thus most likely a bug in Chromium itself and not in Marco.

**Comment by darkgeek** [comment]
@lukefromdc Hi, thanks for your reply. I have disabled the "Use hardware acceleration when available" option in Chromium, restart it and Mate desktop, but nothing improved.

**Comment by monsta** [comment]
@vkareh: could it be related to XPresent/DRI3 support somehow...?

**Comment by vkareh** [comment]
It could be, although XPresent lets marco to use the GPU for compositing (so based on @lukefromdc's comment, it wouldn't be due to XPresent) . XPresent is only enabled if the driver supports it, so we could try to rule it out just in case.

@darkgeek - what video card (and which driver) are you using?

Maybe related, but I've seen something similar in the Slack desktop app, where by default it runs with the `--disable-gpu` flag and I get awful tearing and choppy scrolling. Once I remove the flag (i.e. allow it to use the GPU for rendering) it works just fine. This is not related to XPresent, as I can see the issue without it.

I'll have to test Chromium to see if I can reproduce this.

**Comment by darkgeek** [comment]
@vkareh My laptop is Samsung Chromebook Plus, it has Rockchip's RK3399 soc inside, which has `Mali T860MP4 GPU` inside. Currently, I have ArchLinuxArm installed on this laptop following their [guide](https://archlinuxarm.org/platforms/armv8/rockchip/samsung-chromebook-plus), and only LLVMpipe is available now:

```shell
justin@alarm:~$ glxinfo | grep render
direct rendering: Yes
    GLX_MESA_multithread_makecurrent, GLX_MESA_query_renderer, 
    GLX_MESA_multithread_makecurrent, GLX_MESA_query_renderer, 
Extended renderer info (GLX_MESA_query_renderer):
OpenGL renderer string: llvmpipe (LLVM 5.0, 128 bits)
    GL_ARB_conditional_render_inverted, GL_ARB_conservative_depth, 
    GL_NV_conditional_render, GL_NV_depth_clamp, GL_NV_packed_depth_stencil, 
    GL_ARB_conditional_render_inverted, GL_ARB_conservative_depth, 
    GL_NV_blend_square, GL_NV_conditional_render, GL_NV_depth_clamp, 
    GL_OES_element_index_uint, GL_OES_fbo_render_mipmap, 
```

**Comment by vkareh** [comment]
hmm...I'm not familiar with this hardware, so I wonder if XPresent is compatible or not... if you run marco from the command line, do you get a message about XPresent?

turn compositing back on (`gsettings set org.mate.Marco.general compositing-manager true`) and then run `marco --replace &` in your terminal. If you've compiled marco with the XPresent extension, and your system is not compatible, you should get a "XPresent is not compatible with your current system configuration." message.

Also, just checked on the [archlinuxarm package listing](https://archlinuxarm.org/packages/arm/marco) and marco doesn't have libxpresent as a dependency, so unless you explicitly installed this library and compiled with it, it wouldn't even show up.

**Comment by darkgeek** [comment]
@vkareh Hi, I run your commands in terminal and don't see errors:
![image](https://user-images.githubusercontent.com/1317617/36434368-d7a09fee-1699-11e8-8535-fc2dcc49ed2b.png)
 
And as to libxpresent dependency, I checked [archlinux x86_64 package listing](https://www.archlinux.org/packages/community/x86_64/marco/) and doesn't see libxpresent also.

**Comment by vkareh** [comment]
@darkgeek - thanks for testing this :)

No error means that you're either running XPresent just fine, or that it's just not there (I guess we should've put a message if XPresent was running properly instead :stuck_out_tongue:). It's most likely that you just don't have XPresent support.

So we're back to Chromium... I'm looking into this and I'm running out of ideas :disappointed: 
Does the issue happen with Compiz as well?

**Comment by darkgeek** [comment]
@vkareh Haven't tested it with Compiz, but tested on Xfce4 with compositing enabled, where chromium works fine there. For chromium, I submited a bug report on their [site](https://bugs.chromium.org/p/chromium/issues/detail?id=813740&q=&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified) :)

**Comment by monsta** [comment]
You can also try testing with Metacity.

**Comment by monsta** [comment]
> I can confirm no such issue exists when I use Xfce4 or Mate before 1.20.

Which version was it before? 1.18, 1.16...?

**Comment by darkgeek** [comment]
@monsta Hi, I use 1.18 before.

**Comment by darkgeek** [comment]
@monsta @vkareh Hi, I'm back😁. Just tried `metacity`, and found that Chromium scrolls smoothly without tearing.

**Comment by muktupavels** [comment]
@darkgeek when you test other window manager please include also versions, in this case `metacity --version`.

**Comment by flexiondotorg** [comment]
It doesn't look like Marco in Arch Linux is built with libxpresent https://www.archlinux.org/packages/community/x86_64/marco/

**Comment by vkareh** [comment]
yeah, so it's somehow related to the marco compositor, but not to XPresent. I'm looking into the code and nothing comes up as a culprit. I'm not able to reproduce this.

Some interaction quirk between Chromium, Arch on Arm, and the Marco compositor seems to be causing this...

**Comment by darkgeek** [comment]
And my `metacity` version is:
```shell
justin@alarm:~$ metacity --version
metacity 3.26.0
Copyright (C) 2001-2009 Havoc Pennington, Red Hat, Inc., and others
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
